docs: Update MySQL Server Version (#24517)
gravitational/teleport: docs/pages/database-access/troubleshooting.mdx

This example is intentionally simple; we could have configured Alice's permissions using more fine-grained control.
For more detailed information about database access controls and how to restrict
access see the [RBAC](../database-access/rbac.mdx) documentation.

> 

Reviewer: the troubleshooting.mdx is included in all kinds of databases. not sure if it will be confusing showing MySQL database error there or if there an easier way to hide/show this (scope?)

Author: The `database-access/troubleshooting.mdx` is used only in pages/database-access/getting-started.mdx:(!docs/pages/includes/database-access/aws-troubleshooting.mdx!) and 
```
     {
          "title": "Troubleshooting",
          "slug": "/database-access/troubleshooting/"
        }
```
```
rg 'troubleshooting.mdx' | grep -v "kubernetes-access" | grep -v "access-controls" | grep -v "desktop-access" | grep -v "deploy-a-cluster" | grep -v "machine-id"
pages/management/admin.mdx:- [Troubleshooting](./admin/troubleshooting.mdx): Collect metrics and diagnostic information from Teleport.
pages/database-access/introduction.mdx:documentation](./troubleshooting.mdx) for common problems and solutions.
pages/database-access/getting-started.mdx:(!docs/pages/includes/database-access/aws-troubleshooting.mdx!)
pages/database-access/guides/postgres-redshift.mdx:(!docs/pages/includes/database-access/aws-troubleshooting.mdx!)
pages/database-access/guides/rds.mdx:(!docs/pages/includes/database-access/aws-troubleshooting.mdx!)
pages/database-access/guides/redis-aws.mdx:(!docs/pages/includes/database-access/aws-troubleshooting.mdx!)
pages/database-access/guides/azure-redis.mdx:(!docs/pages/includes/database-access/azure-troubleshooting.mdx!)
pages/database-access/guides/azure-postgres-mysql.mdx:(!docs/pages/includes/database-access/azure-troubleshooting.mdx!)
pages/database-access/guides/rds-proxy.mdx:(!docs/pages/includes/database-access/aws-troubleshooting.mdx!)
pages/database-access/guides/redshift-serverless.mdx:(!docs/pages/includes/database-access/aws-troubleshooting.mdx!)

where other docs integration for most cases refers to `aws-troubleshooting.mdx!`

## Connection to MySQL database results in "Unknown system variable 'query_cache_size'" error

> When a Teleport Proxy is configured with Teleport Proxy Listener Mode (`proxy_listener_mode: separate`) by default MySQL client connections to the Teleport Proxy don't use the TLS Routing feature that allows propagating correct MySQL Server Version.

Reviewer 2: I think the structure of this sentence pretty ambiguous. Does this mean that TLS Routing allows propagating the correct MySQL server version, or that, by not using TLS Routing, the Proxy Service can propagate the correct version?

Author: @ptgott 
Yes, With TLS Routing enabled the MySQL Engine version propagation works correctly and this  issue occurs only for setups where TLS Routing is disabled. 

I have rephrased this sentence to: 
> When TLS Routing feature is disable by default, the Teleport Proxy Service returns `8.0.0-Teleport` as the MySQL server version. In some cases, like connecting with a GUI Client, this can result in obtaining an `Unknown system variable 'query_cache_size'` error that indicates that MySQL capabilities were not properly negotiated between the MySQL client and server.

> By default, Teleport Proxy returns `8.0.0-Teleport` as the MySQL Server version. When the default MySQL Version is returned. In some cases like connecting with GUI Client, this can result in obtaining an `Unknown system variable 'query_cache_size'` error that indicates that MySQL Capabilities were not properly negotiated between MySQL Client and Server.

Reviewer 2: ```suggestion
By default, the Teleport Proxy Service returns `8.0.0-Teleport` as the MySQL server version. In some cases, like connecting with a GUI Client, this can result in obtaining an `Unknown system variable 'query_cache_size'` error that indicates that MySQL capabilities were not properly negotiated between the MySQL client and server.
```

> One way to solve this issue is to use the TLS Routing feature [TLS Routing Migration](https://goteleport.com/docs/manageAment/operations/tls-routing/) where correct MySQL Server Version is propagation via TLS Routing extensions.

Reviewer 2: ```suggestion
One way to solve this issue is to [use the TLS Routing feature](https://goteleport.com/docs/manageAment/operations/tls-routing/), where the Teleport Proxy Service propagates the correct MySQL server version via TLS Routing extensions.
```

> Another way to bypass this error where migration to TLS Routing is not possible is to use the `tsh proxy db --tunnel <database-name>` [Teleport Local Proxy Command](https://goteleport.com/docs/connect-your-client/gui-clients/#get-connection-information) that allows establishing TLS Routing connection to the Teleport Proxy even if TLS Routing was not enabled on the Teleport Cluster.

Reviewer 2: ```suggestion
If migration to TLS Routing is not possible, another way to bypass this error is to use the [Teleport local proxy command](https://goteleport.com/docs/connect-your-client/gui-clients/#get-connection-information), which allows you to establish a TLS Routing connection to the Teleport Proxy Service even if TLS Routing was not enabled on the Teleport cluster.
```

> Alternately if you don't want to migrate to TLS Routing mode and use the Teleport Local Proxy you can overwrite the default MySQL Server Version (8.0.0-Teleport) returned by Teleport Proxy by settings the `mysql_server_version` in the `proxy_service` Teleport Proxy configuration:

Reviewer 2: ```suggestion
Another possibility is to overwrite the default MySQL server version (8.0.0-Teleport) returned by the Teleport Proxy Service. To do this, assign the `mysql_server_version` field in the `proxy_service` configuration block on your Teleport Proxy Service instances:
```

access see the [RBAC](../database-access/rbac.mdx) documentation.

## Connection to MySQL database results in "Unknown system variable 'query_cache_size'" error

> When TLS Routing feature is disable by default, the Teleport Proxy Service returns `8.0.0-Teleport` as the MySQL server version. In some cases, like connecting with a GUI Client, this can result in obtaining an `Unknown system variable 'query_cache_size'` error that indicates that MySQL capabilities were not properly negotiated between the MySQL client and server.

Reviewer 2: ```suggestion

When TLS Routing is disable by default, the Teleport Proxy Service returns `8.0.0-Teleport` as the MySQL server version. In some cases, like connecting with a GUI Client, this can result in obtaining an `Unknown system variable 'query_cache_size'` error that indicates that MySQL capabilities were not properly negotiated between the MySQL client and server.
```
- Let's standardize on an empty line between a heading and the first paragraph after it
- Grammar tweak
